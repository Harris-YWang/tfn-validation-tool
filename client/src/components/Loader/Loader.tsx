import React from 'react';
import { ILoaderProps } from '../../types';

const Loader = ({ showLoader }: ILoaderProps) =>
	showLoader ? (
		<div className="m-3">
			<div className="spinner-border text-primary" role="status" />
			<span className="m-2">Loading...</span>
		</div>
	) : null;

export default Loader;
