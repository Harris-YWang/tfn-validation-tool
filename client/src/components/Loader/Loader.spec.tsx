import { shallow } from 'enzyme';
import React from 'react';

import Loader from './Loader';

describe('Loader Component', () => {
	it('should render when showLoader is true', () => {
		const showLoader = true;
		const wrapper = shallow(<Loader showLoader={showLoader} />);

		expect(wrapper).toMatchSnapshot();
	});

	it('should render nothing when showLoader is false', () => {
		const showLoader = false;
		const wrapper = shallow(<Loader showLoader={showLoader} />);

		expect(wrapper.text()).toEqual('');
	});
});
