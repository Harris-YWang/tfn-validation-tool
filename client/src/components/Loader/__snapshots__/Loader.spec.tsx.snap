// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component should render when showLoader is true 1`] = `ShallowWrapper {}`;
